Feature/oauth auto account linking

Baguette-bbang opened the pull request:
## :hash: 연관된 이슈
#482

## :memo: 작업 내용
동일한 이메일로 Apple과 Kakao 등 여러 OAuth 제공자에 가입할 때 발생하는 중복 이메일 제약조건 위반 문제를 해결하기 위해 자동 계정 연결 기능을 구현

동일 이메일로 여러 OAuth 제공자 가입 시 자동 계정 통합 처리합니다.

Baguette-bbang committed:
feat: OAuth 자동 계정 연결 서비스 추가

- OAuthAccountLinkingService: 동일 이메일로 여러 OAuth 제공자 가입 시 자동 계정 통합
- OAuthAccountLinkingServiceTest: 자동 계정 연결 로직 단위 테스트
- 기존 사용자 조회 및 새로운 OAuth 제공자 연결 처리
- ThreadLocal을 활용한 계정 연결 상태 관리

Baguette-bbang committed:
feat: OAuth 서비스에 자동 계정 연결 로직 통합

- AppleOAuthService: 자동 계정 연결 서비스 활용하여 기존 사용자와 연결
- KakaoOAuthService: 자동 계정 연결 서비스 활용하여 기존 사용자와 연결
- OAuthService: 웹/앱 로그인 방식 통합 및 자동 계정 연결 처리
- 동일 이메일로 여러 OAuth 제공자 가입 시 자동 계정 통합 기능

Baguette-bbang committed:
feat: 사용자 관련 기능 개선

- UserEntity: getAuths() → getAuthEntities() 메소드명 변경
- UserRepository: OAuth 제공자별 사용자 조회 메소드 추가
  - findByEmailAndProvider(): 특정 OAuth 제공자로 가입된 사용자 조회
  - findOAuthProvidersByEmail(): 사용자의 OAuth 제공자 목록 조회
- UserServiceImpl: getAuths() → getAuthEntities() 호출 수정

Baguette-bbang committed:
feat: OAuth API 및 응답 개선

- OAuthController: 테스트용 API 엔드포인트 추가
- LoginResponse: OAuth 계정 연결 정보 필드 추가
  - accountLinkingOccurred: 계정 자동 연결 발생 여부
  - linkedProviders: 연결된 OAuth 제공자 목록
- ErrorCode: OAuth 계정 연결 관련 에러 코드 정리
  - 사용자 확인 관련 불필요한 에러 코드 제거

1117mg reviewed (approved):
테스트코드 작성까지 👍 고생하셨습니다!

ddhi7 reviewed (approved):
자동 계정 연결 기능..👍 고생하셨습니다

Baguette-bbang committed:
refactor: OAuthAccountLinkingService를 메소드 분리 및 Entity 반환 방식으로 리팩토링

- ThreadLocal 제거하고 명시적 반환값 사용
- findOrCreateUser()와 linkOAuthProviderToUser() 메소드로 분리
- AccountLinkingResult DTO 제거하고 UserEntity 직접 반환
- AppleOAuthService, KakaoOAuthService에서 새로운 구조 적용
- OAuthService에서 Repository 직접 사용하도록 수정
- 단위 테스트를 새로운 구조에 맞게 수정
- 불필요한 단순 위임 메소드들 제거

Baguette-bbang commented on src/main/java/org/swyp/dessertbee/auth/service/OAuthAccountLinkingService.java:
```diff
@@ -0,0 +1,177 @@
+     * 계정 연결 상태 초기화 (요청 처리 완료 후 호출)
+     */
+    public void resetAccountLinkingStatus() {
+        accountLinkingOccurred.set(false);
```
해당 코드 제거햇습니다! 감사합니다
